cumulus/polkadot-parachain/src/common/command.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Cumulus.
+
+// Cumulus is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Cumulus is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Cumulus.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Helper trait used to dynamically run parachain node commands.
+
+use cumulus_client_cli::ExportGenesisHeadCommand;
+use cumulus_primitives_core::BlockT;
+use frame_benchmarking_cli::BlockCmd;
+#[cfg(any(feature = "runtime-benchmarks"))]
+use frame_benchmarking_cli::StorageCmd;
+use sc_block_builder::BlockBuilderApi;
+use sc_cli::{CheckBlockCmd, ExportBlocksCmd, ExportStateCmd, ImportBlocksCmd, RevertCmd};
+use sc_client_api::{BlockBackend, StorageProvider, UsageProvider};
+use sc_client_db::{Backend, DbHash};
+use sc_service::{Configuration, PartialComponents, TaskManager};
+use sp_api::ProvideRuntimeApi;
+use sp_blockchain::HeaderBackend;
+use sp_runtime::{traits::Header as HeaderT, OpaqueExtrinsic};
+use std::{fmt::Debug, future::Future, pin::Pin, str::FromStr};
+
+type SyncCmdResult = sc_cli::Result<()>;
+type AsyncCmdResult<'a> = (Pin<Box<dyn Future<Output = SyncCmdResult> + 'a>>, TaskManager);
+
+pub trait CmdRunner<Block: BlockT> {
```

Comment:
```suggestion
pub(crate) trait CmdRunner<Block: BlockT> {
```

Reply:
I ended up removing this trait and struct and adding other ones. I used `pub(crate)` for all of them.